flow/layers/image_filter_layer.cc

```diff
@@ -39,12 +39,22 @@ void ImageFilterLayer::Diff(DiffContext* context, const Layer* old_layer) {
 void ImageFilterLayer::Preroll(PrerollContext* context,
                                const SkMatrix& matrix) {
   TRACE_EVENT0("flutter", "ImageFilterLayer::Preroll");
+
   Layer::AutoPrerollSaveLayerState save =
       Layer::AutoPrerollSaveLayerState::Create(context);
 
+  context->raster_cached_entries.emplace_back(RasterCacheEntry(this));
+  auto current_index = context->raster_cached_entries.size();
+
+  auto& cache_entry = context->raster_cached_entries.back();
+
   SkRect child_bounds = SkRect::MakeEmpty();
   PrerollChildren(context, matrix, &child_bounds);
 
+  cache_entry.num_child_entries =
+      context->raster_cached_entries.size() - current_index;
+  cache_entry.need_caching = IsNeedCached(context, matrix);
```

Comment:
Here we should distinguish between caching the layer and caching children of the layer.